android/app/src/main/kotlin/org/getlantern/lantern/util/LanternHttpClient.kt

```diff
@@ -93,9 +91,27 @@ open class LanternHttpClient : HttpClient() {
                 ) {
                     Logger.debug(TAG, "JSON response" + result.toString())
                     result?.let {
-                        val user = JsonUtil.fromJson<ProUser>(result.toString())
-                        Logger.debug(TAG, "User ID is ${user.userId}")
-                        cb.onSuccess(response!!, user)
+                        try {
+                            val user = JsonUtil.fromJson<ProUser>(result.toString())
+                            Logger.debug(TAG, "User ID is ${user.userId}")
+                            cb.onSuccess(response!!, user)
+                        } catch (e: SerializationException) {
+                            // If for some reason we get error return user with basic string
+                            Logger.error(TAG, "Unable to parse user from JSON", e)
```

Comment:
So I found that is you pass devices as null then you still crash, So I added this, so even if we get an exception we catch it and then return some basic details, So at least the app does not crash.

Reply:
To test this, if I take the user data JSON response and set devices to null, I get another exception in the catch block (it looks like the email is missing):

```
07-04 07:21:42.640  1501  1790 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.google.gson.JsonElement.getAsString()' on a null object reference
07-04 07:21:42.640  1501  1790 E AndroidRuntime: 	at org.getlantern.lantern.model.LanternHttpClient$userData$1.onSuccess(LanternHttpClient.kt:104)
07-04 07:21:42.640  1501  1790 E AndroidRuntime: 	at org.getlantern.lantern.model.LanternHttpClient$proRequest$1.onResponse(LanternHttpClient.kt:321)
07-04 07:21:42.640  1501  1790 E AndroidRuntime: 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
07-04 07:21:42.640  1501  1790 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
07-04 07:21:42.640  1501  1790 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
07-04 07:21:42.640  1501  1790 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:1012)
```

Reply:
@jigar-f What do you think of this change to add and make use of a tryParseJson utility method that catches and logs any exception? https://github.com/getlantern/lantern-client/pull/1117